subprojects/dependency-management/src/main/java/org/gradle/internal/locking/DefaultDependencyLockingProvider.java

```diff
@@ -42,9 +43,11 @@
     private final boolean writeLocks;
     private final boolean partialUpdate;
     private final LockEntryFilter lockEntryFilter;
+    private final DomainObjectContext context;
 
-    public DefaultDependencyLockingProvider(FileResolver fileResolver, StartParameter startParameter, boolean isProjectScope) {
-        this.lockFileReaderWriter = new LockFileReaderWriter(fileResolver, isProjectScope);
+    public DefaultDependencyLockingProvider(FileResolver fileResolver, StartParameter startParameter, DomainObjectContext context, boolean isProjectScope) {
+        this.context = context;
+        this.lockFileReaderWriter = new LockFileReaderWriter(fileResolver, context, isProjectScope);
```

Comment:
Is it now possible to use `DomainObjectContext.isScript()` or `DomainObjectContext.getProjectPath() == null` in place of `isProjectScope`? It would be good to get rid of the nasty `projectScope` boolean from the services registry?

Reply:
You are right, `DomainObjectContext.isScript()` gives me the information I need.